wiki/markdown/generate-service.md

```diff
@@ -0,0 +1,617 @@
+
+```
+ event-repository.enabled : <true | false>
+```
+
+**NOTE:** Allowed value for _event-repository.enabled_ is _true/false_.If the value is not one of this the service will be terminated.
+
+If _event-repository.enabled_ is false then REMReM Generate will not Lookup ER data. Refer below snippet for instance,
+
+```
+            event-repository.enabled : false
+            event-repository.url     : http(s)://<host>:<port>/<context-path>
+```
+
+If _event-repository.enabled_ is true then Event Repository URL should be mandatory.
+
+```
+            event-repository.enabled : true
+            event-repository.url     : http(s)://<host>:<port>/<context-path>
+```
+
+**NOTE:** If Lookup configurations are enabled and EventRepository URL is not provided then the remrem service gets terminated.
+
+```
+            event-repository.enabled : true
+            event-repository.url     :
+```
+
+## Usage
+
+Available REST resources for REMReM Generate Service are described below:
+
+<table style="width:100%">
+
+<tbody>
+
+<tr>
+
+<th>Resource</th>
+
+<th>Method</th>
+
+<th>Parameters</th>
+
+<th style="margin:0px;">Request body</th>
+
+<th width="20%">Description</th>
+
+</tr>
+
+<tr>
+
+<td>/mp</td>
+
+<td>POST</td>
+
+<td>
+
+mp - message protocol, required
+
+msgType - Eiffel event type, required
+
+failIfMultipleFound - default: false
+
+failIfNoneFound - default: false
+
+</td>
+
+<td>
+
+<pre>{
+  "msgParams": {
+    "meta": {
+      # Matches the meta object
+    }
+  },
+  "eventParams": {
+    "data": {
+      # Matches the data object
+    },
+    "links": {
+      # Matches the links object
+    }
+  }
+}</pre>
+
+</td>
+
+<td>
+
+This endpoint is used to generate Eiffel events and then the obtained event could be published by [Eiffel REMReM Publish](https://github.com/eiffel-community/eiffel-remrem-publish).
+
+</td>
+
+</tr>
+
+<tr>
+
+<td>/event_types/{mp}</td>
+
+<td>GET</td>
+
+<td>
+
+mp - message protocol, required
+
+</td>
+
+<td></td>
+
+<td>This endpoint is used to obtain Eiffel event types implemented in [Eiffel REMReM Semantics](https://github.com/eiffel-community/eiffel-remrem-semantics).</td>
+
+</tr>
+
+<tr>
+
+<td>/template/{type}/{mp}</td>
+
+<td>GET</td>
+
+<td>
+
+type - Eiffel event type
+
+mp - message protocol, required
+
+</td>
+
+<td></td>
+
+<td>This endpoint is used to obtain Eiffel event templates implemented in [Eiffel REMReM Semantics](https://github.com/eiffel-community/eiffel-remrem-semantics).</td>
+
+</tr>
+
+<tr>
+
+<td>/versions</td>
+
+<td>GET</td>
+
+<td></td>
+
+<td></td>
+
+<td>This endpoint is used to get versions of generate service and all loaded protocols versions in JSON format.</td>
+
+</tr>
+
+</tbody>
+
+</table>
+
+## Examples
+
+Typical examples of usage Eiffel REMReM Generate Service endpoints are described below.
+
+You can use command line tools like [curl](https://en.wikipedia.org/wiki/CURL) or some plugin for your favorite browser. For example:
```

Comment:
Not really sure wikipedia should be used as reference here - isn't it better to use their own documentation? https://curl.haxx.se/